### PR TITLE
Add Project Files

### PR DESCRIPTION
**Pull request by AbdelrhmanWalaa**

### CODE REVIEW

**Comment by AmrElAbd09** [comment]
Delivery video Link:
https://drive.google.com/file/d/18fT2M_iGboBTeThPvK7_me9P3f5lazBn/view?usp=drive_link


We identified a minor bug in our system that had NO impact on CPU calculations, task periodicity, or execution times. The issue caused a slight shifting in the timeline of tasks on the logic analyzer, resulting in undesired behavior with the hyperperiod. However, we have successfully resolved this simple bug, and here is the video link for it:
https://drive.google.com/file/d/1yWK4wNTq9Qg-Z8uWGe54AafGlbFJ0Wee/view?usp=sharing

File: Files/tasks.c [hunk 1]
```diff
@@ -0,0 +1,5533 @@
...
+						currently executing task. */
+						/*...NEW... */
+						/* as soon as a new task is available in the EDF ready list,
+						a context switching should take place. Modify preemption way as any task
+						with a sooner deadline must preempt a task with a larger deadline
+						instead of a priority
+						*/
+						#if (configUSE_EDF_SCHEDULER == 1)
+						{
+							if( pxTCB->xTaskPeriod <= pxCurrentTCB->xTaskPeriod )
```
Comment (Hala-Abdelkader) [inline Files/tasks.c:2954]: Here you have to make a condition that the task with the sooner deadline must preempt the task with a larger deadline

Note: -Deadline is stored in a variable named "xItemValue"

Comment (AbdelrhmanWalaa) [reply Files/tasks.c:2954]: Ok, we'll modify it.

Comment (hazemashrafali) [reply Files/tasks.c:2954]: thank you @Hala-Abdelkader 
done, you can check it

**Pull request merged**